pydantic/main.py

```diff
@@ -632,7 +632,12 @@ def copy(
         )
 
         # new `__fields_set__` can have unset optional fields with a set value in `update` kwarg
-        if update:
+        extra_ignore = self.__config__.extra == Extra.ignore
```

Comment:
you also need to take care of `forbid` - either by excluding or raising an error.

Reply:
Probably better to raise an error (like it would if called with `__init__`)